core/tauri-codegen/src/context.rs

```diff
@@ -504,16 +492,70 @@ pub fn context_codegen(data: ContextData) -> Result<TokenStream, EmbeddedAssetsE
   }))
 }
 
-fn ico_icon<P: AsRef<Path>>(
+pub fn icon_image_codegen(path: &Path) -> EmbeddedAssetsResult<TokenStream> {
+  let out_dir = ensure_our_dir()?;
+  let mut segments = Punctuated::new();
+  segments.push(PathSegment {
+    ident: Ident::new("tauri", Span::call_site()),
+    arguments: PathArguments::None,
+  });
+  let root = syn::Path {
+    leading_colon: Some(Token![::](Span::call_site())),
+    segments,
+  };
+  image_icon(&root.to_token_stream(), &out_dir, path)
+}
+
+fn ensure_our_dir() -> EmbeddedAssetsResult<PathBuf> {
```

Comment:
my bad I suggested a wrong name it should be 
```suggestion
fn ensure_out_dir() -> EmbeddedAssetsResult<PathBuf> {
```

Can we also move this function above `icon_image_codegen`

Reply:
I'll move all of them to a new file later on, working on nsis utils right now